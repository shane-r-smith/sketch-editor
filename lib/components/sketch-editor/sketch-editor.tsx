import Stack from "@mui/material/Stack";
import type { PropsWithChildren } from "react";

export function SketchEditor({
  children,
  ...rest
}: PropsWithChildren<React.HTMLAttributes<HTMLElement>>) {
  return (
    <Stack
      flex={1}
      position="relative"
      overflow="hidden"
      width="100%"
      height="100%"
      maxWidth="100vw"
      maxHeight="100vw"
      {...rest}
    >
      <Stack
        flex={1}
        flexDirection="column"
        position="relative"
        overflow="hidden"
        width="100%"
        height="100%"
      >
        {children}
      </Stack>
    </Stack>
  );
}
