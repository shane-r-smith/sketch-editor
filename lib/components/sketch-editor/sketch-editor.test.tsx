import { render } from "@testing-library/react";
import { SketchEditor } from "./sketch-editor";

describe("<SketchEditor />", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <SketchEditor>
        <div>Workspace</div>
      </SketchEditor>
    );
    expect(container).toMatchSnapshot();
  });
});
