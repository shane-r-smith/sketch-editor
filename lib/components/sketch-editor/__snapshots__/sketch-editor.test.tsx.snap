// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SketchEditor /> > matches snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1kbb7q6-MuiStack-root"
  >
    <div
      class="MuiStack-root css-7hwbe1-MuiStack-root"
    >
      <div>
        Workspace
      </div>
    </div>
  </div>
</div>
`;
